app/components/task/multiple-extension-form.js

```diff
@@ -0,0 +1,172 @@
+        } catch (error) {
+          state.set('error', error.message);
+          state.set('isLoading', false);
+          console.error(error);
+          this.toast.error(error.message, '', {
+            ...toastNotificationTimeoutOptions,
+            timeOut: '3000',
+          });
+        }
+      }
+    })();
+
+    return state;
+  });
+
+  get extensionData() {
+    const result = {};
+    result['isLoading'] = this.load.get('isLoading');
+    result['value'] = this.load.get('value');
+    result['error'] = this.load.get('error');
+    return result;
+  }
+
+  @action
+  createNewExtensionRequest() {
+    this.createExtensionRequest = true;
+  }
+
+  @action
+  closeForm() {
+    this.createExtensionRequest = false;
+  }
+
+  @action
+  async submitExtensionRequest(e) {
+    e.preventDefault();
+    this.disableExtensionRequestClose = true;
+    this.createExtensionRequestError = null;
+    //submit button
+    e.submitter.disabled = true;
+    const formData = new FormData(e.target);
+    const extensionTime = new Date(formData.get('newEndsOn')).getTime() / 1000;
+    const json = {};
+    formData.forEach(function (value, key) {
+      json[key] = value;
+    });
+
+    if (extensionTime < this.args.task.endsOn) {
+      this.toast.error(WARNING_INVALID_NEW_ETA, '', {
+        ...toastNotificationTimeoutOptions,
+        timeOut: '3000',
+      });
+      e.submitter.disabled = false;
+      this.disableExtensionRequestClose = false;
+      this.createExtensionRequestError =
+        'The newEndsOn value cannot be smaller than the oldEndsOn value';
+      return;
+    }
+    json['newEndsOn'] = extensionTime;
+    //setting default values
+    json['taskId'] = this.args.task.id;
+    json['assignee'] = this.userState.get('id');
+    json['oldEndsOn'] = this.args.task.endsOn;
+    json['status'] = 'PENDING';
+
+    try {
+      const response = await fetch(
+        `${ENV.BASE_API_URL}/extension-requests?dev=true`,
+        {
+          credentials: 'include',
+          method: 'POST',
+          body: JSON.stringify(json),
+          headers: {
+            'Content-Type': 'application/json',
+          },
+        }
+      );
+      const data = await response.json();
+      if (data.message === 'Extension Request created successfully!') {
+        this.disableExtensionRequestClose = false;
+        this.toast.success(data.message, '', {
+          ...toastNotificationTimeoutOptions,
+          timeOut: '3000',
+        });
+        setTimeout(this.args.closeModel, 2000);
+        return;
+      }
+      this.toast.error('Something went wrong!', '', {
+        ...toastNotificationTimeoutOptions,
+        timeOut: '3000',
+      });
+      e.submitter.disabled = false;
+    } catch (error) {
+      this.toast.error(error.message, '', {
+        ...toastNotificationTimeoutOptions,
+        timeOut: '3000',
+      });
+      setTimeout(this.args.closeModel, 2000);
+    }
+  }
+
+  @action
+  changeExtensionRequestETA(e) {
```

Comment:
e.target.value property is the value of the input element that the event occurred on. In this case, it is the newEndsOn input element. However, the code is setting the e.target.value property to an empty string even if the extensionTime is valid. This means that the user will lose their input if the new ETA is valid.

Reply:
Done. Removed that empty string now user is not losing data.